### PR TITLE
Add a fixture reduction and redaction script

### PR DESCRIPTION
**Pull request by simonheimlicher**
Opening as a draft because it adds tooling rather than fixing a defect, so the shape is open to direction.

## What

`scripts/make-detection-fixture.py` performs steps 3, 6, and 7 of the capture-and-promotion procedure the corpus README already documents: it rejects a capture whose `.data.source` is not `detection`, reduces the screen to the same tail `agentDetectionRecentLines` reads, and applies redactions without changing any line's visible width.

## Why

Those three steps are mechanical, and the width rule is the one that breaks a fixture without looking broken. The agent CLIs wrap, shorten, and truncate rows to the terminal width, so a redaction one character longer than what it replaces moves a wrap point. The fixture then pins a screen the CLI would never render, and it pins it convincingly, because nothing about the file looks wrong.

The padding goes into the spaces immediately after the replacement rather than at the end of the row. That keeps every later column where the capture put it:

```text
│   <ACCOUNT_ID_0000>'s Org                          │ /release-notes for more    │
```

Absorbing the two-character difference at the end of the line instead preserves the line's length while sliding the closing border two columns left. I made exactly that mistake while writing this, and the regeneration check below is what caught it.

A replacement that cannot fit is refused rather than applied:

```
error: line 5: replacing 'Claude Max' with 'Claude Maximum Subscription Plan' widens
the row by 22 column(s), and only 1 space(s) follow it; every column after this point
would shift. Choose a replacement of 11 characters or fewer
```

## Evidence

Regenerating three committed fixtures from their original captures reproduces each byte for byte:

```
MATCH  676-wrapped-background-agent-wait   (claude 2.1.226, 36 columns)
MATCH  676-background-agent-wait           (claude 2.1.224)
MATCH  676-compound-elapsed-status         (claude 2.1.224)
```

Those three come from #691 and are not on `main` yet, so the check runs against that branch. As a negative control, feeding a `--source screen` capture exits 2 and writes nothing, and the widening case above exits 1 and writes nothing.

## Scope

The script chooses no redactions of its own beyond masking dollar amounts, which `--keep-money` disables. What to redact stays step 7's decision and the author's. Tool versions: Python 3.14.6, `prowl read --source detection --json` from a 2026.8.9 Debug build.

### CODE REVIEW

**Review by onevclaw — CHANGES_REQUESTED** [review]
I found two blocking cases where the generator violates its visible-width guarantee:

- Money redaction bypasses `substitute()` and always replaces a matched amount with `$X.XX`. For example, `$123.45` becomes `$X.XX`, reducing the line width by two columns and shifting any trailing border.
- `substitute()` uses Python `len()`, which counts code points rather than terminal cells. Replacing `猫` with `X`, for example, reduces the displayed width by one column even though both strings have length 1. The same issue can affect combining characters and emoji.

These cases can produce fixtures with layouts that the captured terminal never displayed. Please make money redaction width-preserving and either use terminal-cell-width calculations compatible with the captured output or reject unsupported non-ASCII replacements and document that limitation.

Please also add generator-level tests covering multi-digit amounts and Unicode replacements or rejection. The current CI passes, but it only validates the committed fixtures and does not execute this script, so it does not exercise these guarantees.

onevclaw - an assistant to @onevcat

File: scripts/make-detection-fixture.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        # a closing box border stays where the capture put it. Past that glyph
+        # the padding carries nothing, and the corpus stores lines without it.
+        out_lines.append(line.rstrip())
+
+    sys.stdout.write("\n".join(out_lines))
+
+    if applied:
+        print("redactions applied (for the metadata file):", file=sys.stderr)
+        for old, new in applied.items():
+            print(f'  "{old} replaced with {new}"', file=sys.stderr)
```
[the remark this answers is not in the document]
Comment (simonheimlicher) [reply scripts/make-detection-fixture.py]: Correct, and this was the worst of the three: the leak ran through the tool whose job was to prevent it. Step 5 sends the author to this summary, so a redacted home path or prompt returned to the repository inside the metadata file.

Fixed in `731557d7`. The summary is now keyed by replacement and counts lines:

```
replacements applied, by line count. Name what each one replaced in the metadata
file — the originals are not printed here, because this summary is committed and
they were the reason to redact:
  1 line → "… replaced with /Users/usr"
```

`test_summary_never_names_what_it_replaced` asserts the original string is absent from stderr, so the property is held by a test rather than by the shape of a format string. The README now says the summary names the replacement only and that describing the original is the author's job.

File: scripts/make-detection-fixture.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            raise WidthError(
+                f"{character!r} (U+{code:04X}) has no width terminals agree on, so the "
+                f"replacement cannot be shown to preserve the row. Use a replacement "
+                f"without joiners or variation selectors"
+            )
+        if code < 0x20 or code == 0x7F:
+            raise WidthError(f"control character U+{code:04X} has no visible width")
+        if unicodedata.combining(character):
+            continue
+        width += 2 if unicodedata.east_asian_width(character) in ("W", "F") else 1
```
[the remark this answers is not in the document]
Comment (simonheimlicher) [reply scripts/make-detection-fixture.py:119]: Right, and it is the same class of error the review caught in the first round: summing parts where the terminal renders a whole. A skin-tone modifier is wide by `east_asian_width` and adds no cell of its own, so `👩🏽` measured four and the row moved by two.

Refused rather than measured, in `731557d7` — `U+1F3FB–1F3FF` and the tag range `U+E0001–E007F` join the joiners and variation selectors:

```
UNDECIDABLE_SPANS = [
    range(0xFE00, 0xFE10),    # variation selectors
    range(0xE0100, 0xE01F0),  # variation selectors supplement
    range(0x1F3FB, 0x1F400),  # emoji modifiers (skin tone)
    range(0xE0001, 0xE0080),  # tag characters, including flag tag sequences
]
```

Matching Ghostty's grapheme width would mean carrying a width table and keeping it in step with the terminal, and being wrong there is silent. Refusing costs an author one rephrased replacement and fails at generation time. Tests cover both the modifier and a flag tag sequence.

File: scripts/make-detection-fixture.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+                    f"replacing {old!r} with {new!r} widens the row by {delta} column(s), "
+                    f"and only {gap} space(s) follow it; every column after this point "
+                    f"would shift. Choose a replacement of {cell_width(old) + gap} cells or fewer"
+                )
+            tail = tail[delta:]
+        rest = tail
+
+
+# Money amounts carry no signal for the classifier and are account data.
+MONEY = re.compile(r"\$\d+\.\d\d")
```
[the remark this answers is not in the document]
Comment (simonheimlicher) [reply scripts/make-detection-fixture.py]: Confirmed, and the silence is the part that matters: the README documents the masking as automatic, so an author who reads that and does not re-read the row commits the amount believing it was handled. A custom status line on this fork already prints a running session cost, and a long-lived session reaches three figures, so grouping is a matter of time rather than a hypothetical.

Both halves of your suggestion, in `731557d7`. Grouped digits mask with the separator kept, which preserves width the same way the ungrouped form does:

```python
MONEY = re.compile(r"\$\d{1,3}(?:,\d{3})+\.\d\d|\$\d+\.\d\d")  # $1,234.56 → $X,XXX.XX
AMOUNT_SHAPED = re.compile(r"\$\d[\d,]*(?:\.\d+)?")
```

Anything dollar-shaped that `MONEY` did not consume now fails the run with a message pointing at `--redact` or `--keep-money`. A bare `$1` is exempt: shell positionals appear in real captures, so a decimal point or a grouping comma is required before a token counts as an amount.

Worth flagging what this feature has cost: it is the only redaction the script applies on its own authority, it has produced two of the five defects found in review, and no committed fixture has ever contained an amount. The README (`c6368f8d`) now names the whole status-line counter family in step 7 — cost, tokens, quota, elapsed — and says the generator covers the dollar and the author owns the rest, since automating one counter of five reads as coverage. I would also accept deleting the masking outright and routing amounts through `--redact` like everything else; say the word and it comes out.
